### ISSUE TITLE
LIGHT THEME: When Portfolio Position set to 10, Out of Bound Exception error

### ISSUE THREAD

**Issue by KathJohnson** [meetinghouse/cms#252]
While testing the portfolio position setting, I found that it worked fine when I set it to a lower number like 1 or 2, but then I set it to 10 to see what would happen and got an error. I am unable to back out of this, actually, and get back to the site, so I can't continue testing on the menus site. Can this be fixed shortly so testing can continue? Thanks.

![screenshot 2016-05-10 at 9 47 36 am](https://cloud.githubusercontent.com/assets/5922040/15148650/4076c08c-1694-11e6-879b-0c8311e2d1dd.png)

**Comment by jboz62** [comment]
I set it to 4 on the New England Soundproofing site, that worked out fine. Did you run the deploy script before testing? There were some other changes you might not have.

**Comment by KathJohnson** [comment]
I just tried the same on the lightthemetest site, which I think is up-to-date. The portfolios position works well until I get to higher numbers. Unfortunately, the Out of Bounds exception error keeps me out of the site so I can't back anything out.

Position 4 was fine, by the way.

**Comment by jboz62** [comment]
It may be the Sudhir was assuming it would always be in one of the top positions, as we asked for it to default to 2 or 3. Seems like that should be at least in somewhere in the top position, maybe 3-5. 

Sudhir, is this issue already fixed or is this a bug that needs fixing?

**Comment by meetinghouse** [comment]
I would be fine with limiting the integers in the position drop-down to
only those which can be used. Or allowing it to be in any position works
too. Whichever is easiest.

**Comment by 1sp** [comment]
Yes john, the problem is that selected portfolio position is greater than
available number of menu items in this particular site. I can push a quick
fix which would put portfolio at the very end of the menu if its position
is invalid. Would that work ?

On Tue, May 10, 2016 at 7:41 PM, John M. Corbett <notifications@github.com>
wrote:

> I would be fine with limiting the integers in the position drop-down to
> only those which can be used. Or allowing it to be in any position works
> too. Whichever is easiest.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/meetinghouse/cms/issues/252#issuecomment-218169527>
>



-- 
Sudhir Porwal

**Comment by jboz62** [comment]
I’m not sure if that will work. Currently, I’m having to put the “all projects” menu item at the end and then hiding it. So if you put it at the end it will disappear. If you can make it so “all projects” never shows up on the menu, then it would work. You can see what I did on this site: newenglandsoundproofing.corbettresearchgroup.com 

Katherine, do you see any issue with that?

**Comment by meetinghouse** [comment]
Why is it necessary to set the All Projects item to be last? Can't we just
uncheck the option of making it a menu item?

**Comment by 1sp** [comment]
Ok, I understand the situation now but we have to choose a position where portfolio can reside if it's configured position turn invalid at any point of time. 

Portfolio position's invalidity can happen on run time too. Think about a scenario where portfolio is at 3rd and total number of menu pages 4. Removing two parent pages from this setup will make portfolio's position invalid (3rd position in a list of two).

Making the logic such that portfolio takes first or last position in the menu in case of invalid position would make the logic work in all possible scenarios.

**Comment by jboz62** [comment]
Oh, I didn’t realize that was an option. That’s great, sorry for confusion.

**Comment by jboz62** [comment]
Yes, that sounds good, I didn’t know we could disable “all projects”, thought that was dynamically generated. I’ll just disable it and you can do that quick fix as described. Thank you.

**Comment by meetinghouse** [comment]
It sounds like this will work and be pretty easy. A win for all.

>

**Comment by jboz62** [comment]
Hi Sudhir, we discovered that one of your migrations ended up being deployed to the S+H site, which is a live site. It has messed up the menu (please see here: http://shconstruction2.corbettresearchgroup.com/ <http://shconstruction2.corbettresearchgroup.com/>) I believe we can roll this back to the previous state, do you know how to do that. I can send you the credentials for Forge if you need that, not sure how this was deployed. This is the current deploy script, maybe you could just edit it and we can paste it in? Not sure if we need to rollback just the migration, or the commit as well.

CUSTOM_NAME=shconstruction2
BRANCH=blog

date

cd /home/forge/$CUSTOM_NAME.corbettresearchgroup.com
git checkout -f $BRANCH
git pull origin $BRANCH
composer install
php artisan migrate
bower install

This is the deploy result:


Fri May  6 17:31:55 EDT 2016
Already on 'blog'
Your branch is ahead of 'origin/blog' by 183 commits.
  (use "git push" to publish your local commits)
From github.com:meetinghouse/cms
 * branch            blog       -> FETCH_HEAD
   c6918f9..ef179a5  blog       -> origin/blog
Merge made by the 'recursive' strategy.
 .../2016_05_06_212515_migrate_old_menu_names.php   | 28 ++++++++++++++++++++++
 1 file changed, 28 insertions(+)
 create mode 100644 app/database/migrations/2016_05_06_212515_migrate_old_menu_names.php
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
Warning: Ambiguous class resolution, "PostsController" was found in both "/home/forge/shconstruction2.corbettresearchgroup.com/app/controllers/PostsContrtoller.php" and "/home/forge/shconstruction2.corbettresearchgroup.com/app/controllers/PostsController.php", the first will be used.
> php artisan clear-compiled
> php artisan optimize
Generating optimized class loader
> php artisan cms:setup
Migrated: 2016_05_06_212515_migrate_old_menu_names
Thank you.

**Comment by 1sp** [comment]
no problem this can be quickly undone. Can you please quickly run `git reflog show` and paste here the output ?

**Comment by jboz62** [comment]
Here it is:

99ce8d0 HEAD@{0}: pull origin blog: Merge made by the 'recursive' strategy.
f6e72b6 HEAD@{1}: checkout: moving from blog to blog
f6e72b6 HEAD@{2}: pull origin blog: Merge made by the 'recursive' strategy.
50e8fbf HEAD@{3}: checkout: moving from blog to blog
50e8fbf HEAD@{4}: pull origin blog: Merge made by the 'recursive' strategy.
e4f4272 HEAD@{5}: checkout: moving from blog to blog
e4f4272 HEAD@{6}: pull origin blog: Merge made by the 'recursive' strategy.
008022c HEAD@{7}: checkout: moving from blog to blog
008022c HEAD@{8}: pull origin blog: Merge made by the 'recursive' strategy.
0c81e3c HEAD@{9}: checkout: moving from blog to blog
0c81e3c HEAD@{10}: pull origin blog: Merge made by the 'recursive' strategy.
da7f324 HEAD@{11}: checkout: moving from blog to blog
da7f324 HEAD@{12}: pull origin blog: Merge made by the 'recursive' strategy.
01d2e06 HEAD@{13}: checkout: moving from blog to blog
01d2e06 HEAD@{14}: pull origin blog: Merge made by the 'recursive' strategy.
6b12172 HEAD@{15}: checkout: moving from blog to blog
6b12172 HEAD@{16}: checkout: moving from blog to blog
6b12172 HEAD@{17}: pull origin blog: Merge made by the 'recursive' strategy.
45962b2 HEAD@{18}: checkout: moving from blog to blog
45962b2 HEAD@{19}: checkout: moving from blog to blog
45962b2 HEAD@{20}: pull origin blog: Merge made by the 'recursive' strategy.
b20cd05 HEAD@{21}: checkout: moving from blog to blog
b20cd05 HEAD@{22}: pull origin blog: Merge made by the 'recursive' strategy.
:...skipping...
99ce8d0 HEAD@{0}: pull origin blog: Merge made by the 'recursive' strategy.
f6e72b6 HEAD@{1}: checkout: moving from blog to blog
f6e72b6 HEAD@{2}: pull origin blog: Merge made by the 'recursive' strategy.
50e8fbf HEAD@{3}: checkout: moving from blog to blog
50e8fbf HEAD@{4}: pull origin blog: Merge made by the 'recursive' strategy.
e4f4272 HEAD@{5}: checkout: moving from blog to blog
e4f4272 HEAD@{6}: pull origin blog: Merge made by the 'recursive' strategy.
008022c HEAD@{7}: checkout: moving from blog to blog
008022c HEAD@{8}: pull origin blog: Merge made by the 'recursive' strategy.
0c81e3c HEAD@{9}: checkout: moving from blog to blog
0c81e3c HEAD@{10}: pull origin blog: Merge made by the 'recursive' strategy.
da7f324 HEAD@{11}: checkout: moving from blog to blog
da7f324 HEAD@{12}: pull origin blog: Merge made by the 'recursive' strategy.
01d2e06 HEAD@{13}: checkout: moving from blog to blog
01d2e06 HEAD@{14}: pull origin blog: Merge made by the 'recursive' strategy.
6b12172 HEAD@{15}: checkout: moving from blog to blog
6b12172 HEAD@{16}: checkout: moving from blog to blog
6b12172 HEAD@{17}: pull origin blog: Merge made by the 'recursive' strategy.
45962b2 HEAD@{18}: checkout: moving from blog to blog
45962b2 HEAD@{19}: checkout: moving from blog to blog
45962b2 HEAD@{20}: pull origin blog: Merge made by the 'recursive' strategy.
b20cd05 HEAD@{21}: checkout: moving from blog to blog
b20cd05 HEAD@{22}: pull origin blog: Merge made by the 'recursive' strategy.
683ef6e HEAD@{23}: checkout: moving from blog to blog
683ef6e HEAD@{24}: pull origin blog: Merge made by the 'recursive' strategy.
b3426fa HEAD@{25}: checkout: moving from blog to blog
b3426fa HEAD@{26}: pull origin blog: Merge made by the 'recursive' strategy.
cc85408 HEAD@{27}: checkout: moving from blog to blog
cc85408 HEAD@{28}: checkout: moving from blog to blog
:...skipping...
99ce8d0 HEAD@{0}: pull origin blog: Merge made by the 'recursive' strategy.
f6e72b6 HEAD@{1}: checkout: moving from blog to blog
f6e72b6 HEAD@{2}: pull origin blog: Merge made by the 'recursive' strategy.
50e8fbf HEAD@{3}: checkout: moving from blog to blog
50e8fbf HEAD@{4}: pull origin blog: Merge made by the 'recursive' strategy.
e4f4272 HEAD@{5}: checkout: moving from blog to blog
e4f4272 HEAD@{6}: pull origin blog: Merge made by the 'recursive' strategy.
008022c HEAD@{7}: checkout: moving from blog to blog
008022c HEAD@{8}: pull origin blog: Merge made by the 'recursive' strategy.
0c81e3c HEAD@{9}: checkout: moving from blog to blog
0c81e3c HEAD@{10}: pull origin blog: Merge made by the 'recursive' strategy.
da7f324 HEAD@{11}: checkout: moving from blog to blog
da7f324 HEAD@{12}: pull origin blog: Merge made by the 'recursive' strategy.
01d2e06 HEAD@{13}: checkout: moving from blog to blog
01d2e06 HEAD@{14}: pull origin blog: Merge made by the 'recursive' strategy.
6b12172 HEAD@{15}: checkout: moving from blog to blog
6b12172 HEAD@{16}: checkout: moving from blog to blog
6b12172 HEAD@{17}: pull origin blog: Merge made by the 'recursive' strategy.
45962b2 HEAD@{18}: checkout: moving from blog to blog
45962b2 HEAD@{19}: checkout: moving from blog to blog
45962b2 HEAD@{20}: pull origin blog: Merge made by the 'recursive' strategy.
b20cd05 HEAD@{21}: checkout: moving from blog to blog
b20cd05 HEAD@{22}: pull origin blog: Merge made by the 'recursive' strategy.
683ef6e HEAD@{23}: checkout: moving from blog to blog
683ef6e HEAD@{24}: pull origin blog: Merge made by the 'recursive' strategy.
b3426fa HEAD@{25}: checkout: moving from blog to blog
b3426fa HEAD@{26}: pull origin blog: Merge made by the 'recursive' strategy.
cc85408 HEAD@{27}: checkout: moving from blog to blog
cc85408 HEAD@{28}: checkout: moving from blog to blog
cc85408 HEAD@{29}: checkout: moving from blog to blog
cc85408 HEAD@{30}: pull origin blog: Merge made by the 'recursive' strategy.
94445d3 HEAD@{31}: checkout: moving from blog to blog
94445d3 HEAD@{32}: pull origin blog: Merge made by the 'recursive' strategy.
fa220ab HEAD@{33}: checkout: moving from blog to blog
fa220ab HEAD@{34}: pull origin blog: Merge made by the 'recursive' strategy.
:...skipping...
99ce8d0 HEAD@{0}: pull origin blog: Merge made by the 'recursive' strategy.
f6e72b6 HEAD@{1}: checkout: moving from blog to blog
f6e72b6 HEAD@{2}: pull origin blog: Merge made by the 'recursive' strategy.
50e8fbf HEAD@{3}: checkout: moving from blog to blog
50e8fbf HEAD@{4}: pull origin blog: Merge made by the 'recursive' strategy.
e4f4272 HEAD@{5}: checkout: moving from blog to blog
e4f4272 HEAD@{6}: pull origin blog: Merge made by the 'recursive' strategy.
008022c HEAD@{7}: checkout: moving from blog to blog
008022c HEAD@{8}: pull origin blog: Merge made by the 'recursive' strategy.
0c81e3c HEAD@{9}: checkout: moving from blog to blog
0c81e3c HEAD@{10}: pull origin blog: Merge made by the 'recursive' strategy.
da7f324 HEAD@{11}: checkout: moving from blog to blog
da7f324 HEAD@{12}: pull origin blog: Merge made by the 'recursive' strategy.
01d2e06 HEAD@{13}: checkout: moving from blog to blog
01d2e06 HEAD@{14}: pull origin blog: Merge made by the 'recursive' strategy.
6b12172 HEAD@{15}: checkout: moving from blog to blog
6b12172 HEAD@{16}: checkout: moving from blog to blog
6b12172 HEAD@{17}: pull origin blog: Merge made by the 'recursive' strategy.
45962b2 HEAD@{18}: checkout: moving from blog to blog
45962b2 HEAD@{19}: checkout: moving from blog to blog
45962b2 HEAD@{20}: pull origin blog: Merge made by the 'recursive' strategy.
b20cd05 HEAD@{21}: checkout: moving from blog to blog
b20cd05 HEAD@{22}: pull origin blog: Merge made by the 'recursive' strategy.
683ef6e HEAD@{23}: checkout: moving from blog to blog
683ef6e HEAD@{24}: pull origin blog: Merge made by the 'recursive' strategy.
b3426fa HEAD@{25}: checkout: moving from blog to blog
b3426fa HEAD@{26}: pull origin blog: Merge made by the 'recursive' strategy.
cc85408 HEAD@{27}: checkout: moving from blog to blog
cc85408 HEAD@{28}: checkout: moving from blog to blog
cc85408 HEAD@{29}: checkout: moving from blog to blog
cc85408 HEAD@{30}: pull origin blog: Merge made by the 'recursive' strategy.
94445d3 HEAD@{31}: checkout: moving from blog to blog
94445d3 HEAD@{32}: pull origin blog: Merge made by the 'recursive' strategy.
fa220ab HEAD@{33}: checkout: moving from blog to blog
fa220ab HEAD@{34}: pull origin blog: Merge made by the 'recursive' strategy.
63513fb HEAD@{35}: checkout: moving from blog to blog
:...skipping...
99ce8d0 HEAD@{0}: pull origin blog: Merge made by the 'recursive' strategy.
f6e72b6 HEAD@{1}: checkout: moving from blog to blog
f6e72b6 HEAD@{2}: pull origin blog: Merge made by the 'recursive' strategy.
50e8fbf HEAD@{3}: checkout: moving from blog to blog
50e8fbf HEAD@{4}: pull origin blog: Merge made by the 'recursive' strategy.
e4f4272 HEAD@{5}: checkout: moving from blog to blog
e4f4272 HEAD@{6}: pull origin blog: Merge made by the 'recursive' strategy.
008022c HEAD@{7}: checkout: moving from blog to blog
008022c HEAD@{8}: pull origin blog: Merge made by the 'recursive' strategy.
0c81e3c HEAD@{9}: checkout: moving from blog to blog
0c81e3c HEAD@{10}: pull origin blog: Merge made by the 'recursive' strategy.
da7f324 HEAD@{11}: checkout: moving from blog to blog
da7f324 HEAD@{12}: pull origin blog: Merge made by the 'recursive' strategy.
01d2e06 HEAD@{13}: checkout: moving from blog to blog
01d2e06 HEAD@{14}: pull origin blog: Merge made by the 'recursive' strategy.
6b12172 HEAD@{15}: checkout: moving from blog to blog
6b12172 HEAD@{16}: checkout: moving from blog to blog
6b12172 HEAD@{17}: pull origin blog: Merge made by the 'recursive' strategy.
45962b2 HEAD@{18}: checkout: moving from blog to blog
45962b2 HEAD@{19}: checkout: moving from blog to blog
45962b2 HEAD@{20}: pull origin blog: Merge made by the 'recursive' strategy.
b20cd05 HEAD@{21}: checkout: moving from blog to blog
b20cd05 HEAD@{22}: pull origin blog: Merge made by the 'recursive' strategy.
683ef6e HEAD@{23}: checkout: moving from blog to blog
683ef6e HEAD@{24}: pull origin blog: Merge made by the 'recursive' strategy.
b3426fa HEAD@{25}: checkout: moving from blog to blog
b3426fa HEAD@{26}: pull origin blog: Merge made by the 'recursive' strategy.
cc85408 HEAD@{27}: checkout: moving from blog to blog
cc85408 HEAD@{28}: checkout: moving from blog to blog
cc85408 HEAD@{29}: checkout: moving from blog to blog
cc85408 HEAD@{30}: pull origin blog: Merge made by the 'recursive' strategy.
94445d3 HEAD@{31}: checkout: moving from blog to blog
94445d3 HEAD@{32}: pull origin blog: Merge made by the 'recursive' strategy.
fa220ab HEAD@{33}: checkout: moving from blog to blog
fa220ab HEAD@{34}: pull origin blog: Merge made by the 'recursive' strategy.
63513fb HEAD@{35}: checkout: moving from blog to blog
63513fb HEAD@{36}: pull origin blog: Merge made by the 'recursive' strategy.
096dc10 HEAD@{37}: checkout: moving from blog to blog
:...skipping...
99ce8d0 HEAD@{0}: pull origin blog: Merge made by the 'recursive' strategy.
f6e72b6 HEAD@{1}: checkout: moving from blog to blog
f6e72b6 HEAD@{2}: pull origin blog: Merge made by the 'recursive' strategy.
50e8fbf HEAD@{3}: checkout: moving from blog to blog
50e8fbf HEAD@{4}: pull origin blog: Merge made by the 'recursive' strategy.
e4f4272 HEAD@{5}: checkout: moving from blog to blog
e4f4272 HEAD@{6}: pull origin blog: Merge made by the 'recursive' strategy.
008022c HEAD@{7}: checkout: moving from blog to blog
008022c HEAD@{8}: pull origin blog: Merge made by the 'recursive' strategy.
0c81e3c HEAD@{9}: checkout: moving from blog to blog
0c81e3c HEAD@{10}: pull origin blog: Merge made by the 'recursive' strategy.
da7f324 HEAD@{11}: checkout: moving from blog to blog
da7f324 HEAD@{12}: pull origin blog: Merge made by the 'recursive' strategy.
01d2e06 HEAD@{13}: checkout: moving from blog to blog
01d2e06 HEAD@{14}: pull origin blog: Merge made by the 'recursive' strategy.
6b12172 HEAD@{15}: checkout: moving from blog to blog
6b12172 HEAD@{16}: checkout: moving from blog to blog
6b12172 HEAD@{17}: pull origin blog: Merge made by the 'recursive' strategy.
45962b2 HEAD@{18}: checkout: moving from blog to blog
45962b2 HEAD@{19}: checkout: moving from blog to blog
45962b2 HEAD@{20}: pull origin blog: Merge made by the 'recursive' strategy.
b20cd05 HEAD@{21}: checkout: moving from blog to blog
b20cd05 HEAD@{22}: pull origin blog: Merge made by the 'recursive' strategy.
683ef6e HEAD@{23}: checkout: moving from blog to blog
683ef6e HEAD@{24}: pull origin blog: Merge made by the 'recursive' strategy.
b3426fa HEAD@{25}: checkout: moving from blog to blog
b3426fa HEAD@{26}: pull origin blog: Merge made by the 'recursive' strategy.
cc85408 HEAD@{27}: checkout: moving from blog to blog
cc85408 HEAD@{28}: checkout: moving from blog to blog
cc85408 HEAD@{29}: checkout: moving from blog to blog
cc85408 HEAD@{30}: pull origin blog: Merge made by the 'recursive' strategy.
94445d3 HEAD@{31}: checkout: moving from blog to blog
94445d3 HEAD@{32}: pull origin blog: Merge made by the 'recursive' strategy.
fa220ab HEAD@{33}: checkout: moving from blog to blog
fa220ab HEAD@{34}: pull origin blog: Merge made by the 'recursive' strategy.
63513fb HEAD@{35}: checkout: moving from blog to blog
63513fb HEAD@{36}: pull origin blog: Merge made by the 'recursive' strategy.
096dc10 HEAD@{37}: checkout: moving from blog to blog
096dc10 HEAD@{38}: pull origin blog: Merge made by the 'recursive' strategy.
56732b1 HEAD@{39}: checkout: moving from blog to blog
56732b1 HEAD@{40}: pull origin blog: Merge made by the 'recursive' strategy.
6ecfed4 HEAD@{41}: checkout: moving from blog to blog
6ecfed4 HEAD@{42}: pull origin blog: Merge made by the 'recursive' strategy.

**Comment by 1sp** [comment]
Please run `git reset --hard HEAD@{1}` and confirm if it take the code to how it was before. If possible paste here the output. Thanks

**Comment by 1sp** [comment]
Please run the above command in "/home/forge/shconstruction2.corbettresearchgroup.com"

**Comment by jboz62** [comment]
That one didn’t seem to make any difference, here’s what it said. I’m running these via SSH in Terminal:

HEAD is now at f6e72b6 Merge branch 'blog' of github.com:meetinghouse/cms into blog
forge@corbettresearchgroup-one:~/shconstruction2.corbettresearchgroup.com$

**Comment by 1sp** [comment]
Interesting. Let's see this together and get this fixed. https://join.me/107-376-466

**Comment by jboz62** [comment]
https://join.me/157-738-741

**Comment by jboz62** [comment]
https://join.me/534-070-147

**Comment by jboz62** [comment]
Did you get the new link? https://join.me/534-070-147

**Comment by 1sp** [comment]
Just pushed the changes to https://github.com/meetinghouse/cms/tree/1sp-in. Please review.

**Comment by jboz62** [comment]
Hi Sudhir,

I reviewed the changes, the page save works now, so the menus can be edited. However, I still don’t see a way to attach blog posts to a single menu item. Maybe there’s a way to create a page that, like the portfolio menu item, will just list the blog posts?

Maybe we need to get S+H working again the way it was so we can figure out how that worked. As I mentioned before, there was a menu item called “builder’s notebook” listed all the posts on one page and in the sub-menu. I don’t know how that worked, exactly, but I’m sure it’s in the code. 

We can’t fix the site without that functionality. So the only option is to try and restore it, I think.

**Comment by jboz62** [comment]
Btw, Katherine sent me a new password, but it doesn’t work either. I’ll send it to your email address and you can try it.

Does this statement from the Resync notes "An ssh keypair has been created with root’s account so root can transfer files without having to enter a password.” mean that “root’s account” is John Corbett’s account, the account owner? I’m not sure what account that is referring to. If it’s John’s account, we should be able to re-key it, right?

**Comment by meetinghouse** [comment]
In casting a wide net today to find help in logging into the Rsync account,
I sent an e-mail to Rsync. They sent this back, which may be the root of
the problem you're having logging in with the password Dan provided, John:

"We have reset your password to the original one that was sent to you in
your welcome email - if you no longer have your welcome email, let us know
and we will resend the password itself to you."

So we are probably back to the password I sent earlier today.

**Comment by jboz62** [comment]
Hmm, interesting. I’ll give that a try. Thanks for the heads up!

**Comment by jboz62** [comment]
Yes, that works, now we just need to locate the backup.

**Comment by jboz62** [comment]
Katherine, I haven’t really gotten anywhere with this. Was able to sftp in, and I downloaded the 2 production databases, but they are from May 6th, so probably the same as what’s in there. I tried installing sshfs to use that to do a restore, but can’t get past the password on that, which is odd as it should be the same password, I think. I tried running this command: `sshfs usw-s007.rsync.net: /root/restore` but I get a `fuse: bad mount point` error on /root/restore. Not sure what that path should be. 

Sudhir if you have any ideas on this, the password is now back to the original one we showed you. Let me know if you don’t have it. thank you.

**Comment by 1sp** [comment]
I am able to ftp in. I am now copying the backup copy of code to compare it with the latest code - will keep updating the progress here

**Comment by jboz62** [comment]
Where did you see the backups, I only saw the current backup in the “forge” directory, but that is, I think, the same as what’s on the forge server. It seems we need to use ssh or sshfs to do this right and be able to access all 7 backups.

**Comment by 1sp** [comment]
Yes, you are right. But first I am going compare the server code with the code in our repository to see if there are any new files on the server for this site (which could be responsible for 'builder's notebook functionality').

I would also analyze the database to see where that builder's notebook's menu entry is. If it was a normal page then the menu entry would still be intact

**Comment by jboz62** [comment]
Sudhir, I thought the issue wasn’t the code, but rather, the database. We already rolled back the commit, so the code should be good. We need to find the right database and replace it, correct?

**Comment by jboz62** [comment]
Ok, thanks. I checked the admin before these changes happened as I was wondering how the “builder’s notebook” worked. There wasn’t a page for that, so I can verify that for you. In fact, I couldn’t find anything labelled “builder’s notebook” which makes me believe it was hard coded somewhere.

**Comment by jboz62** [comment]
Sudhir, do you use Google Drive?

**Comment by 1sp** [comment]
yes, via the email id I shared with you yesterday

**Comment by jboz62** [comment]
Ok, thanks, I'll have Katherine send you the Google Doc link with instructions on accessing via ssh and sshfs, although, I had no luck getting it to work for me. The password wasn't working, using the original pass, not the one on the doc.

**Comment by 1sp** [comment]
I am finally able to ssh it and check the quota etc. Trying to run other commands mentioned in the documentation. https://join.me/531-399-071

**Comment by 1sp** [comment]
Now able to see the snapshots as well :)

**Comment by jboz62** [comment]
Wow, awesome! How did you get in? Did you have to add the root key to your setup? I guess Katherine must have sent you the Google Doc?

**Comment by jboz62** [comment]
https://join.me/770-393-605

**Comment by 1sp** [comment]
@jboz62 join me at https://join.me/825-566-035 asap

**Comment by 1sp** [comment]
I think I now understand what is happening with the issue on shconstruction2.corbettresearchgroup.com. I would need to discuss with @jboz62 @KathJohnson . Please join me on https://join.me/825-566-035

**Comment by 1sp** [comment]
Sorry for all the emails - Is anybody around ?

**Comment by meetinghouse** [comment]
I am here. I just clicked the join me link.

**Comment by meetinghouse** [comment]
Sudhir,

John here, not sure how much help I can be but I'd sure like to know - - IF
I can understand.....


Cordially,

John Corbett, President
Corbett Research Group
http://www.restorationtrades.com
"Web Marketing that Works"

Publishers of:

The Custom Building & Restoration Trades Directory
http://www.restorationtradesdirectory.com

The Building Arts Notebook, "Journal of the Guild of Building Artisans"
http://buildingartisansguild.com/

Email:  John@RestorationTrades.com
Phone: 413-586-4748

**Comment by jboz62** [comment]
https://join.me/562-762-151

**Comment by jboz62** [comment]
Let’s try this one: https://join.me/731-255-265

**Comment by 1sp** [comment]
I can only see your screen, call is not going through

On Thu, May 12, 2016 at 1:27 AM, jboz62 <notifications@github.com> wrote:

> Let’s try this one: https://join.me/731-255-265
>
>
> > On May 11, 2016, at 2:50 PM, Sudhir <notifications@github.com> wrote:
> >
> > I think I now understand what is happening with the issue on
> shconstruction2.corbettresearchgroup.com. I would need to discuss with
> @jboz62 <https://github.com/jboz62> @KathJohnson <
> https://github.com/KathJohnson> . Please join me on
> https://join.me/825-566-035 <https://join.me/825-566-035>
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly or view it on GitHub <
> https://github.com/meetinghouse/cms/issues/252#issuecomment-218553847>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/meetinghouse/cms/issues/252#issuecomment-218571742>
>



-- 
Sudhir Porwal

**Comment by jboz62** [comment]
I relaunched join.me app, let’s see if that works: https://join.me/498-416-871

**Comment by jboz62** [comment]
are you still on, Sudhir?

**Comment by 1sp** [comment]
Yes John, can you please a new join me link again?

On Thu, May 12, 2016 at 1:44 AM, jboz62 <notifications@github.com> wrote:

> are you still on, Sudhir?
>
> > On May 11, 2016, at 4:06 PM, John Bozaan <john@bozaandesign.com> wrote:
> >
> > I relaunched join.me <http://join.me/> app, let’s see if that works:
> https://join.me/498-416-871 <https://join.me/498-416-871>
> >
> >> On May 11, 2016, at 4:05 PM, Sudhir <notifications@github.com <mailto:
> notifications@github.com>> wrote:
> >>
> >> I can only see your screen, call is not going through
> >>
> >> On Thu, May 12, 2016 at 1:27 AM, jboz62 <notifications@github.com
> <mailto:notifications@github.com>> wrote:
> >>
> >> > Let’s try this one: https://join.me/731-255-265 <
> https://join.me/731-255-265>
> >> >
> >> >
> >> > > On May 11, 2016, at 2:50 PM, Sudhir <notifications@github.com
> <mailto:notifications@github.com>> wrote:
> >> > >
> >> > > I think I now understand what is happening with the issue on
> >> > shconstruction2.corbettresearchgroup.com <
> http://shconstruction2.corbettresearchgroup.com/>. I would need to
> discuss with
> >> > @jboz62 <https://github.com/jboz62 <https://github.com/jboz62>>
> @KathJohnson <
> >> > https://github.com/KathJohnson <https://github.com/KathJohnson>> .
> Please join me on
> >> > https://join.me/825-566-035 <https://join.me/825-566-035> <
> https://join.me/825-566-035 <https://join.me/825-566-035>>
> >> > > —
> >> > > You are receiving this because you were mentioned.
> >> > > Reply to this email directly or view it on GitHub <
> >> > https://github.com/meetinghouse/cms/issues/252#issuecomment-218553847
> <https://github.com/meetinghouse/cms/issues/252#issuecomment-218553847>>
> >> >
> >> > —
> >> > You are receiving this because you commented.
> >> > Reply to this email directly or view it on GitHub
> >> > <
> https://github.com/meetinghouse/cms/issues/252#issuecomment-218571742 <
> https://github.com/meetinghouse/cms/issues/252#issuecomment-218571742>>
> >> >
> >>
> >>
> >>
> >> --
> >> Sudhir Porwal
> >> —
> >> You are receiving this because you were mentioned.
> >> Reply to this email directly or view it on GitHub <
> https://github.com/meetinghouse/cms/issues/252#issuecomment-218573809>
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/meetinghouse/cms/issues/252#issuecomment-218576332>
>



-- 
Sudhir Porwal

**Comment by jboz62** [comment]
Ah, there you are: https://join.me/701-044-506

**Comment by jboz62** [comment]
Hi Sudhir,

Can you give us an estimate on the 2 items we discussed yesterday before starting? The items were:

1) Add blog (posts) capability to both themes with ability to name menu item. See (builder’s notebook for example)
2) Ability to rename the “portfolios” menu item to “projects” or whatever is needed.

Thank you,
John

**Comment by 1sp** [comment]
Hi John,

For both of the above tasks I would take 2-3 hours. I an start right now.

**Comment by jboz62** [comment]
Ok, thank you, Sudhir. Katherine, do you want him to do this now or wait?

**Comment by meetinghouse** [comment]
Sudhir,

Can you remind me what tasks this is? Just point me to the notes, if you
wish, and I'll get you an answer.


Cordially,

John Corbett, President
Corbett Research Group
http://www.restorationtrades.com
"Web Marketing that Works"

Publishers of:

The Custom Building & Restoration Trades Directory
http://www.restorationtradesdirectory.com

The Building Arts Notebook, "Journal of the Guild of Building Artisans"
http://buildingartisansguild.com/

Email:  John@RestorationTrades.com
Phone: 413-586-4748

**Comment by jboz62** [comment]
If you view the email on github (link at bottom), you will see the email chain and what he is responding to, Katherine.

**Comment by 1sp** [comment]
Any update on this? The original bug in this thread is already fixed (though not deployed I guess).

**Comment by jboz62** [comment]
Katherine, do you have any followup on this email chain? Please view on Github to see all the details.

**Comment by KathJohnson** [comment]
I appreciate the fixing of the bug, but at this moment I am not deploying anything to any live sites until I have a better handle on a process for that.

I do have a question on the bug fix: is it already merged into the blog branch code base?

Also, @1sp,  is your estimate of 2-3 hours for each task separately (for a total of 4-6 hours), or a total for both tasks? 

Many many thanks for all attention to this. It is not going to the dustbin, but for the moment, please don't proceed. When we're ready to go ahead, I'll create separate issues for both requests.

**Comment by 1sp** [comment]
Thanks @KathJohnson , my estimate was total for both tasks as described in https://github.com/meetinghouse/cms/issues/252#issuecomment-218780279

**Comment by jboz62** [comment]
Closing this issue as it seems to no longer be a problem. Just tested light and dark theme and was able to move portfolio to position 10 without issue.
